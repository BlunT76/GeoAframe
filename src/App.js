//react imports
import React, {Component} from 'react';
import './App.css';
//capacitor imports
import {Plugins} from '@capacitor/core';
const {Geolocation, Camera} = Plugins;

class App extends Component {

    constructor(props) {
        super(props);
        this.showDeviceInfo = this
            .showDeviceInfo
            .bind(this);

    }

    state = {
        lat: null,
        lng: null
    }

    async showDeviceInfo() {

        const watch = Geolocation.watchPosition({}, (position, err) => {
          console.log(position)
          if(!err)
          this.setState({lat: position.coords.latitude, lng: position.coords.longitude})
        })

        //async takePicture() {
          const image = await Camera.getPhoto({
            quality: 90,
            allowEditing: true,
            //resultType: CameraResultType.Uri
          });
       
        //console.log('Current', coordinates);
        //await this.setState({lat: coordinates.coords.latitude, lng: coordinates.coords.longitude})
    }

    componentWillUnmount(){
      //Geolocation.clearWatch(watch)
    }
    render() {
        return (
            <div className="App">
                <header className="App-header">
                    <h1>Test react and capacitor</h1>
                    <button onClick={this.showDeviceInfo}>
                        Show Device Info</button>
                    <p>lat: {this.state.lat}</p>
                    <p>lng: {this.state.lng}</p>
                </header>

            </div>
        );
    }
}

export default App;
