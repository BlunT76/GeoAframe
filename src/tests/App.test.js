import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';

const mockGeolocation = {
  getCurrentPosition: jest.fn(),
  watchPosition: jest.fn(),
};
global.navigator.geolocation = mockGeolocation;

const mockAlert = {
  alert: jest.fn(),
};
App.alert = mockAlert;

test('App is always the same', () => {
  const component = renderer.create(
    <App />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
