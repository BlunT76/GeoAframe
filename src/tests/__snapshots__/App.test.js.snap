// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App is always the same 1`] = `
<div>
  <a-scene
    className={Object {}}
    cursor="rayOrigin: mouse"
    embedded={true}
    light="defaultLightsEnabled: false"
    style={
      Object {
        "height": 768,
        "width": 1024,
      }
    }
  >
    <a-entity>
      <a-entity
        camera="active: true"
        far="1000"
        id="camera"
        look-controls="enabled: true"
        near="0.005"
        rotation="0 -180 0"
        userHeight="1.6"
        wasd-controls={true}
      />
    </a-entity>
    <video
      autoPlay={true}
      className="videoContainer"
      height={480}
      muted={false}
      playsInline={true}
      width={640}
    />
    <a-text
      color="red"
      position="-6378136.999961403 -10 0"
      rotation="-90 90 -180"
      value="N"
      width="50"
    />
    <a-text
      color="white"
      position="-6378136.999961403 -10 0"
      rotation="-90 90 0"
      value="S"
      width="50"
    />
    <a-text
      color="white"
      position="-6378136.999961142 -10 22.2638981586095"
      rotation="-90 90 90"
      value="E"
      width="50"
    />
    <a-text
      color="white"
      position="-6378136.999961142 -10 -22.2638981586095"
      rotation="-90 90 -90"
      value="O"
      width="50"
    />
  </a-scene>
  <div />
</div>
`;
